t/lib/overload_join.t

```diff
@@ -0,0 +1,162 @@
+is( $kayo . " zamm", "*kayo* *zamm*", 'BoldStars concatenation' );
+is( "zamm " . $kayo, "*zamm* *kayo*", 'BoldStars concatenation (swapped)' );
+is( "$kayo + $biff", "*kayo* *+* *biff*", 'BoldStars stringification' );
+
+my $obj = "$kayo ";
+is( $obj, '*kayo* ', 'Contatenation with space (right)' );
+isa_ok( $obj, 'BoldStars' );
+
+$obj = " $kayo";
+is( $obj, ' *kayo*', 'Contatenation with space (left)' );
+isa_ok( $obj, 'BoldStars' );
+
+$obj = $kayo . $biff;
+is( "$obj", "*kayobiff*", "BoldStars concatenation" );
+isa_ok( $obj, 'BoldStars' );
+
+# test that join uses concat overload
+{
+    my $expected     = BoldStars->new('kayo biff');
+    my $expected_str = "$expected";
+
+    my $interpolated = "$kayo $biff";
+    is( $interpolated, $expected,     'interpolation (space)' );
+    is( $interpolated, $expected_str, 'interpolation str (space)' );
+    isa_ok( $interpolated, 'BoldStars' );
+
+    my $reduced = reduce { $a . ' ' . $b } $kayo, $biff;
+    is( $reduced, $expected,     'reduce (space)' );
+    is( $reduced, $expected_str, 'reduce str (space)' );
+    isa_ok( $reduced, 'BoldStars' );
+
+    my $joined = join( ' ', $kayo, $biff );
+    is( $joined, $expected,     "join (space)" );
+    is( $joined, $expected_str, "join str (space)" );
+    isa_ok( $joined, 'BoldStars' );
+}
+
+{
+    my $expected     = BoldStars->new('kayo + sock');
+    my $expected_str = "$expected";
+
+    my $interpolated = "$kayo + $sock";
+    is( $interpolated, $expected,     'interpolation (non-space)' );
+    is( $interpolated, $expected_str, 'interpolation str (non-space)' );
+    isa_ok( $interpolated, 'BoldStars' );
+
+    my $reduced = reduce { $a . ' + ' . $b } $kayo, $sock;
+    is( $reduced, $expected,     'reduce (non-space)' );
+    is( $reduced, $expected_str, 'reduce str (non-space)' );
+    isa_ok( $reduced, 'BoldStars' );
+
+    my $joined = join( ' + ', $kayo, $sock );
+    is( $joined, $expected,     "join (non-space)" );
+    is( $joined, $expected_str, "join str (non-space)" );
+    isa_ok( $joined, 'BoldStars' );
+}
+
+# join with a overloaded delim
+{
+    my $expected     = BoldStars->new('ouch + sock');
+    my $expected_str = "$expected";
+
+    my $delim = BoldStars->new(' + ');
+    my $interpolated = "ouch${delim}sock";
+    is( $interpolated, $expected,     'interpolation (overload-delim)' );
+    is( $interpolated, $expected_str, 'interpolation str (overload-delim)' );
+    isa_ok( $interpolated, 'BoldStars' );
+
+    my $reduced = reduce { $a . $delim . $b } 'ouch', 'sock';
+    is( $reduced, $expected,     'reduce (overload-delim)' );
+    is( $reduced, $expected_str, 'reduce str (overload-delim)' );
+    isa_ok( $reduced, 'BoldStars' );
+
+    my $joined = join( $delim, 'ouch', 'sock' );
+    is( $joined, $expected,     "join (overload-delim)" );
+    is( $joined, $expected_str, "join str (overload-delim)" );
+    isa_ok( $joined, 'BoldStars' );
+}
+
+done_testing;
+
+__END__
+
+$s = join $del, @list
```

Comment:
This is a lot of comment for a unit test, that seems like it should be part of an Issue ticket, or ongoing discussion, or something? Or maybe part of a still-draft test that needs to be expanded on?

Either way I'm not sure it wants committing as a real part of the actual test file.